library/edi.inc

```diff
@@ -768,6 +649,459 @@ function getUsernames()
 function arrFormated(&$item, $key)
 {
     $item = strstr($item, '_');
-    $item = substr($item, 1, strlen($item)-1);
-    $item = "'".$item;
+    $item = substr($item, 1, strlen($item) - 1);
+    $item = "'" . $item;
+}
+
+function requestEligibility($partner = '', $x12_270 = '')
+{
+    global $X12info;
+    if (((int)$X12info['id'] !== (int)$partner) && (int)$partner > 0) {
+        $X12info = getX12Partner($partner);
+    }
+
+    $payloadId = "3b8c13f5-11e2-43bf-bc47-737cca04f3fe"; // a default fallback
+    if (function_exists('openssl_random_pseudo_bytes') === true) {
+        $data = openssl_random_pseudo_bytes(16);
```

Comment:
Do you need a cryptographically secure random function here?

Reply:
We definitely need a good GUID and/or UUID function especially for the interop stuff. I use openssl here for the random generator more so than security. I did look to see if we had one but didn't find. I have one I did for FHIR somewhere i'll see if I can get into sanitize lib.

Reply:
For future reference, have a crytographically secure random byte generator here (the crypto.php library is always loaded):
https://github.com/openemr/openemr/blob/master/library/crypto.php#L399

And in the near future (when MFA PR comes into codebase) will have a cool crytographically secure random string generator here:
https://github.com/openemr/openemr/blob/5b8cfd847ecff0117991ab43f51c3f3d24858eb3/library/crypto.php#L413

Reply:
okay, just need to use these and create a function to create a UUID/GUID v4. Were going to need to start moving towards adding a companion UUID to pid for patients. I believe Rod brought this up sometime back. If I ever get back on FHIR this is needed.

Reply:
hi @sjpadgett ,
The produceRandomString() function is now available in the main codebase. Supporting UUID/GUID would be very nice. Note it also would not be tough to add a mechanism in the upgrade script to support a custom method that would create a UUID/GUID for each patient when upgrading openemr.
-brady

Reply:
btw, I'd be happy to make an issue for the patient UUID/GUID to see if a new developer is interested in this (this is moderate difficulty so would be a good second project for a new dev); just let me know if there are any details I need to know.

Reply:
Would be nice to have. UUID/GUID will get used at some point so let somebody have at it.